### ISSUE TITLE
Future directions: Reicast Gold and relationship with open-source

### ISSUE THREAD

**Issue by skmp** [reicast/reicast-emulator#1928]
So, over the past months me, div & kamaro have been working privately on an improved version of reicast. We initially started the work in this repo, however, as more months passed and I had more thoughts on the drama around Me, TwinAphex, and FlyingHead I decided I'd rather not have more of the new work done publicly.

The "new" stuff we have right now is: A new front-end UI, a nearly complete HLE bios implementation that replaces the existing bugfest that reios is, and a new arm7di core (to replace GPL'd vba arm -- and fix a few bugs on the way).

A hot topic during the internal team discussions has been whenever or not we should release the code once we get things released, and if so to which extent.

Myself, I've lost much of my trust on the open source model, and I'm tired of leeches in the scene. And by leeches I mostly mean people involved/leading libretro and flycast.

The situation right now is that some trusted 3rd party devs have access to the private repo, both to be able to learn from the changes we do if they wish to, and to make sure that the work doesn't go to waste if something bad happens to me.

I'm open to add any previous reicast devs I trust, as well as people that want to see the code, but more in a "shared source" aspect rather than "open source", til I test the waters more and my trust in the public scene has recovered.

Meanwhile, we'll be releasing a Gold edition of reicast, to cover part of the costs of having two people working full time, plus infrastructure costs (which thankfully are fairly minimal). Personally, I don't see things breaking even, but at least it'll reduce the financial burden on me.

In the future I want to see a lot new features added, including but not limited to
- Getting rid of all the code flyinghead wrote or touched. I don't want this bad taste in my mouth, nor the shitty code it came with.
- Improved UI
- Multiplayer similar to what nullDC had in implementation, but in a more organized way, likely integrated with discord.
- Curated highres textures, modpacks and achievements, integrated in the emulator UI
- Accelerated versions (Renderscript or opencl or both) of the softrend
- Possibly a vulkan renderer if there's enough benefit
- WinCE support
- Naomi, Naomi 2, and other Dreamcast derivatives
- (that's just me typing a wishlist, so, dunno what else)

For now, the open source / free version will remain as is. Once things are more clear, both in my head, and in code, it's likely we'll modularize some of the Gold only features (HLE bios, and such) and have a split repo where those aren't public, but the core of the emulator is. Maybe.

Yeah, I think that's all for now?

Any Questions / Feedback?

PS.
Before you go "omg not open source source bad bad bad', I'd like to note that this is the 2nd open source dreamcast emulator to go closed source because of TwinAphex's involvement, and maybe you should place your anger towards that team, rather than me.

~raz

**Comment by skmp** [comment]
@reicast/collaborators @reicast/devs @reicast/maintainers Notify list I guess

**Comment by skmp** [comment]
Also @reicast/members

**Comment by Doctorwho1909** [comment]
You do what you feel is right

**Comment by Narann** [comment]
> I'm tired of leeches in the scene. And by leeches I mostly mean people involved/leading libretro and flycast.

Lets call this by its name: libretro is a "leechware": They took everything and don't give a fuck about upstream. This is depressing af and I totally understand our point. Other emulators have been destroyed by this. It's sad for open source, but not working with upstream is a shitty mindset and have huge consequences.

Keep the good work !

**Comment by chyyran** [comment]
Have you considered moving to a "Source-Available but not Open Source" model, a la [Ms-RSL,](https://en.wikipedia.org/wiki/Shared_Source_Initiative#Microsoft_Reference_Source_License_(Ms-RSL)) with the option to go full open source later at your discretion (via a rights-waiver CLA like Near/byuu did with GPLv3 to ISC)? 

Although if the RetroArch team was willing to keep in proprietary Sony code in their codebase before being called out, I can understand reservations about them not respecting a reference-only source available license from the reicast team.

**Comment by skmp** [comment]
Yes, i've thought of it as an option. As i'm fairly salty about the entire thing, I don't want to to even indirectly support the libretro efforts though. For now i'm more positive of just sharing with trusted emudevs - and open up from there.

They are known for scrapping the changelogs of a certain other dreamcast emulator and implementing "bogus" bugfixes that were specifically added there to test if they are doing that.

**Comment by Narann** [comment]
Please don't use those licenses. Instead, choose the good old "bins with sources" model so you can stay GPL.

Here are my points:

* Your emulation efforts are valuable on the long run. If you stop to work on it, your code couldn't be reuse.
* When you will leave your emulator work, someone could take it make revive it. mupen64plus story is this. The main dev is not here, still it's a quiet functional emulator, have been ported to x64, ARM, etc.

So you are a non-community driven emulator, but stiil be GPL opensource. This alow you to say: It's over, here is my last release (bin+source, or source alone), fly and have fun people!

The problem with "source available but not open source" is that switch model is impossible. You can't easily go back, specially if you work with few other peoples.

Please consider this when you choose your license. I totally agree with libretro toxicity, but don't let this letting you take the bad decision on the long run.

My 2cts.

Keep the good work!

**Comment by DimitrisVlachos** [comment]
> Please don't use those licenses. Instead, choose the good old "bins with sources" model so you can stay GPL.
> 
> Here are my points:
> 
>     * Your emulation efforts are valuable on the long run. If you stop to work on it, your code couldn't be reuse.
> 
>     * When you will leave your emulator work, someone could take it make revive it. mupen64plus story is this. The main dev is not here, still it's a quiet functional emulator, have been ported to x64, ARM, etc.
> 
> 
> So you are a non-community driven emulator, but stiil be GPL opensource. This alow you to say: It's over, here is my last release (bin+source, or source alone), fly and have fun people!
> 
> The problem with "source available but not open source" is that switch model is impossible. You can't easily go back, specially if you work with few other peoples.
> 
> Please consider this when you choose your license. I totally agree with libretro toxicity, but don't let this letting you take the bad decision on the long run.
> 
> My 2cts.
> 
> Keep the good work!

You definitely have a point but in that case there are enough people already with access to source code and *if* everything insane happens the emu can still be around. So far with just a few exceptions the emulator had no benefit of it being open source. Contributions were always on the low side and in reality it brought toxicity to the scene and intense emotional and psychological stress to @skmp . It's not easy to handle negative criticism when you DON'T deserve it, and messing your name and popularity in the community for people that have nothing better to do in their life than leeching off others work. Personally if this was my emulator i would use GPL license, wouldn't allow RA to use my code (via legal way) and remove completely the source from the public (it didn't helped anyway). 

I'm not defending @skmp , even if he's a good and old friend of mine, i just defend what is fair and the stress and psychological aspect to a person that this can have. It's not like he's behind a fake username. He has a real name, address, linkedin, business everything so it's hard to handle it.

Things are good as they are now in regards to the emulator. Few & Trusted people can have access to the code. Also the tasks at this point require real and serious work it would not benefit at all from going opensource plus it never really did (with FEW exceptions).

**Comment by Narann** [comment]
> You definitely have a point but in that case there are enough people already with access to source code and if everything insane happens the emu can still be around.

Not if the license does not allow to be reused.

Once again, I understand all the toxicity involved, I know public source brings few valuable contributions. I just speak for the *after*, once the code is here and skmp is not (10 years, 20 years, etc). Emulator code is so valuable it stays for years. Does the license will allow anyone to take the code and continues from it? That's what I'm just asking skmp to think about.

Take care of yourself, but please don't let your work being legally unusable in the years to come.
